packages/ui-extensions-react/src/surfaces/point-of-sale/render.tsx

```diff
@@ -39,22 +39,21 @@ export function reactExtension<Target extends RenderExtensionTarget>(
   return extension<'pos.home.tile.render'>(target as any, async (root, api) => {
     const element = await render(api as ApiForRenderExtension<Target>);
 
-    await new Promise<void>((resolve, reject) => {
+    return new Promise<() => void>((resolve, reject) => {
```

Comment:
I think we can move most of this into a shared util: 
```ts
async function remoteRootRender(node: ReactNode) {
   return new Promise<() => void>((resolve, reject) => {
      try {
        const remoteRoot = createRoot(root);
        remoteRoot.render(node);
        return resolve(remoteRoot.unmount);
      } catch (error) {
        // Workaround for https://github.com/Shopify/ui-extensions/issues/325
        // eslint-disable-next-line no-console
        console.error(error);
        reject(error);
      }
    });
}
```

```tsx
return extension<'pos.home.tile.render'>(target as any, async (root, api) => {
  const element = await render(api as ApiForRenderExtension<Target>);
  return remoteRootRender(
    <ExtensionApiContext.Provider value={api}>
      <ErrorBoundary>{element}</ErrorBoundary>
    </ExtensionApiContext.Provider>
  );
}
```

Reply:
Got it, just wanted to check in with you first before making that larger change. I'll try to change this today if I have any downtime, but likely will be tomorrow.